shared/src/balancer_event_handler.rs

```diff
@@ -224,42 +213,31 @@ impl BalancerPools {
     fn contract_to_balancer_events(
         &self,
         contract_events: Vec<EthContractEvent<ContractEvent>>,
-    ) -> Result<Vec<(EventIndex, BalancerEvent)>> {
+    ) -> Result<Vec<(EventIndex, PoolCreated)>> {
         contract_events
             .into_iter()
-            .filter_map(|EthContractEvent { data, meta }| {
+            .map(|EthContractEvent { data, meta }| {
                 let meta = match meta {
                     Some(meta) => meta,
-                    None => return Some(Err(anyhow!("event without metadata"))),
+                    None => return Err(anyhow!("event without metadata")),
                 };
                 match data {
-                    ContractEvent::PoolRegistered(event) => {
-                        Some(convert_pool_registered(&event, &meta))
-                    }
-                    ContractEvent::TokensRegistered(event) => {
-                        Some(convert_tokens_registered(&event, &meta))
-                    }
-                    ContractEvent::TokensDeregistered(event) => {
-                        tracing::error!("unexpected Token Deregistration event {:?}", event);
-                        None
-                    }
-                    _ => {
-                        // TODO - Not processing other events at the moment.
-                        // https://github.com/gnosis/gp-v2-services/issues/681
-                        None
-                    }
+                    ContractEvent::PoolCreated(event) => convert_pool_created(&event, &meta),
```

Comment:
Still using the match to unpack the wrapped data, but there is only one possibility for this match... Is there something simpler I could do to unpack it?

Reply:
No, you always need the match.

Reply:
Personally I also find the `match` nice in this case as the code guarantees that there is only a single event (if there were more variants, you would get a compiler error).